### ISSUE TITLE
Outputs '# SKIP' on a test that is actually run

### ISSUE THREAD

**Issue by darrinwillis** [python-tap/pytest-tap#23]
In pytest, a 'strict' xfail test is one that is expected to fail, and when it passes, that is considered to be a test failure.
I have the following test

```
import pytest
@pytest.mark.xfail(strict=True)
def test_fail():
    # This errors, which means the test passes
    assert 1 == 2
```

which reports:

```
ok 1 - test_fail # SKIP 
1..1
```

The 'ok' part is correct, but the 'SKIP' is incorrect, since it did in fact run the test. (This can confuse the user, especially when obfuscated by a TAP driver)

Interestingly, the test 

```
import pytest
@pytest.mark.xfail(strict=True)
def test_fail():
    # This test case doesn't error, which means the test should fail
    assert 1 == 1
```

reports:

```
not ok 1 - test_fail
# [XPASS(strict)]
1..1
```

Which is correct and doesn't have the above problem. However, when 'strict=False', it still fails, which isn't correct (the strict flag seems to be ignored in all cases).

**Comment by mblayman** [comment]
Wow. That's understandable albeit bizarre from the underlying data structures in the code. From what I've encountered so far, xfails come back as a report with an outcome of `skipped` which is why the plugin marked them as skipped. I'll need to investigate what pytest is doing with xfails when strict mode is on.

Thanks for another solid report.

**Comment by mblayman** [comment]
@darrinwillis Sorry it took two years to get this fixed! I'm finally at a company that uses pytest and I'm reading through the pytest book by Brian Okken and understanding how to work in a pytest-based test suite.

I corrected the `ok/not ok` mismatch with xfails. I also changed the directive from SKIP to TODO which better aligns with the TAP spec. http://testanything.org/tap-specification.html#todo-tests
